src/test/java/com/thealgorithms/scheduling/PreemptivePrioritySchedulingTest.java

```diff
@@ -2,32 +2,46 @@
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
+import com.thealgorithms.devutils.entities.ProcessDetails;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 import org.junit.jupiter.api.Test;
 
 /**
  * Test Cases of Preemptive Priority Scheduling Algorithm
+ *
```

Comment:
Looks good. But using Parameterized tests will improve a bit of tests, after applying changes it will look something like this:

```
package com.thealgorithms.scheduling;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.thealgorithms.devutils.entities.ProcessDetails;
import java.util.List;
import java.util.stream.Stream;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

/**
 * Test Cases of Preemptive Priority Scheduling Algorithm
 *
 * @author [Bama Charan Chhandogi](https://www.github.com/BamaCharanChhandogi)
 */
class PreemptivePrioritySchedulingTest {

    @ParameterizedTest
    @MethodSource("provideProcessesAndExpectedSchedules")
    void testPreemptivePriorityScheduling(List<ProcessDetails> processes, List<String> expectedSchedule) {
        PreemptivePriorityScheduling scheduler = new PreemptivePriorityScheduling(processes);
        scheduler.scheduleProcesses();
        assertEquals(expectedSchedule, scheduler.ganttChart);
    }

    static Stream<Arguments> provideProcessesAndExpectedSchedules() {
        return Stream.of(
                Arguments.of(
                        List.of(
                                new ProcessDetails("P1", 0, 5, 2),
                                new ProcessDetails("P2", 1, 4, 4),
                                new ProcessDetails("P3", 2, 2, 6),
                                new ProcessDetails("P4", 4, 1, 8)
                        ),
                        List.of("P1", "P2", "P3", "P3", "P4", "P2", "P2", "P2", "P1", "P1", "P1", "P1")
                ),

                Arguments.of(
                        List.of(
                                new ProcessDetails("P1", 2, 5, 3),
                                new ProcessDetails("P2", 5, 3, 5),
                                new ProcessDetails("P3", 7, 1, 9)
                        ),
                        List.of("Idle", "Idle", "P1", "P1", "P1", "P2", "P2", "P3", "P2", "P1", "P1")
                )
        );
    }
}
```
But need to add correct formatting, and maybe reorder imports.

What do you think?

Reply:
I had similar thoughts about using parameterized tests. I hesitated initially because other test classes for scheduling algorithms weren't using them. 

But you're right, it would improve the tests. If you'd like, I can update the PR with the parameterized tests. What do you think?

Reply:
It will be good because parameterized tests will improve code readability, reduce code duplication. Because code becomes more compact, and code is separated from test data. It will be easy of adding new test cases. So please do this.

Reply:
Thank you for the suggestion and guidance. I've implemented the changes as discussed.